Date/MomentFormatConverter.php

```diff
@@ -59,6 +57,42 @@ class MomentFormatConverter
      */
     public function convert($format)
     {
-        return strtr($format, self::$formatConvertRules);
+        $size = strlen($format);
+        $foundOne = false;
+        $output = '';
+        //process the format string letter by letter
+        for ($i = 0; $i < $size; $i++) {
```

Comment:
Why not use `foreach( str_split($format) as $char)`? You would spare computing `$size`…